app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -87,6 +92,13 @@ def reset_attempt_count_if_user_no_longer_locked_out
     )
   end
 
+  def timeframe_expired_event
```

Comment:
My fault for recommending the name, but since this concern is used as a shared mixin within other controllers, I think it could help to disambiguate the name a bit more, to make it clearer which "timeframe" we're referring to:

```suggestion
  def sign_in_notification_timeframe_expired_event
```